### ISSUE TITLE
Unpredictable API

### ISSUE THREAD

**Issue by Ivanca** [swisnl/jQuery-contextMenu#453]
Why the items callbacks (each option in the contex menu) aren't normal jQuery events but instead being called manually? (line 885 `callback.call(root.$trigger, key, root)`) and besides that those are being called with weird arguments, the first parameter of a callback should always be an `Event` object, that's how all event listeners work in jQuery, jQuery.UI and hundreads of plugins, then the other parameters could be the key and the root.

**Comment by Ruud-cb** [comment]
While you are probably right and have a valid argument it SHOULD be changed, I guess it is not really useful to open a issue for it because it CANNOT be changed. Changing the order of arguments would break a lot of applications. The developers have made that decision apparently years ago...

**Comment by Ivanca** [comment]
A major release can break backwards compatibility, that's what they are for; perhaps 3.0 or anything after.

**Comment by bbrala** [comment]
Yeah i know the API is sometimes a bit weird. But i didn't feel like changing the argument order, i really hate breaking backwards compatibility.
